internal/saucecloud/cloud.go

```diff
@@ -0,0 +1,67 @@
+package saucecloud
+
+import (
+	"context"
+	"fmt"
+	"github.com/rs/zerolog/log"
+	"github.com/saucelabs/saucectl/cli/progress"
+	"github.com/saucelabs/saucectl/internal/concurrency"
+	"github.com/saucelabs/saucectl/internal/job"
+	"github.com/saucelabs/saucectl/internal/region"
+	"github.com/saucelabs/saucectl/internal/resto"
+	"github.com/saucelabs/saucectl/internal/storage"
+)
+
+// CloudRunner represents the cloud runner for the Sauce Labs cloud.
+type CloudRunner struct {
+	ProjectUploader storage.ProjectUploader
+	JobStarter      job.Starter
+	JobReader       job.Reader
+	CCYReader       concurrency.Reader
+	Region          region.Region
+	ShowConsoleLog  bool
+}
+
+func (r *CloudRunner) uploadProject(filename string) (string, error) {
+	progress.Show("Uploading project")
+	resp, err := r.ProjectUploader.Upload(filename)
+	progress.Stop()
+	if err != nil {
+		return "", err
+	}
+	log.Info().Str("storageId", resp.ID).Msg("Project uploaded.")
+	return resp.ID, nil
+}
+
+// logSuite display the result of a suite
+func (r *CloudRunner) logSuite(res result) {
+	if res.job.ID == "" {
+		log.Error().Str("suite", res.suiteName).Msgf("failed to start")
+		log.Error().Str("suite", res.suiteName).Msgf("%s", res.err)
+		return
+	}
+	resultStr := "Passed"
+	if !res.job.Passed {
+		resultStr = "Failed"
+	}
+	jobDetailsPage := fmt.Sprintf("%s/tests/%s", r.Region.AppBaseURL(), res.job.ID)
+	log.Info().Str("suite", res.suiteName).Msgf("Status: %s - %s", resultStr, jobDetailsPage)
+	r.logSuiteConsole(res)
+}
+
+// logSuiteError display the console output when tests from a suite are failing
+func (r *CloudRunner) logSuiteConsole(res result) {
+	// To avoid clutter, we don't show the console on job passes.
+	if res.job.Passed || !r.ShowConsoleLog {
```

Comment:
This line here is the only logic change, everything else is copy paste.
The original code had the `ShowConsoleLog` variable be read from the project, rather than the runner.
This was moved so it could be used in a common way.
This may get moved behind an interface further down the line.